components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/EndpointResource.java

```diff
@@ -29,55 +33,77 @@
 import org.wso2.carbon.inbound.endpoint.internal.http.api.APIResource;
 
 import javax.xml.namespace.QName;
+import java.io.IOException;
 import java.util.Collection;
 import java.util.HashSet;
 import java.util.Map;
 import java.util.Objects;
 import java.util.Set;
 
-public class EndpointResource extends APIResource {
+import static org.wso2.micro.integrator.management.apis.Constants.ACTIVE_STATUS;
+import static org.wso2.micro.integrator.management.apis.Constants.INACTIVE_STATUS;
+import static org.wso2.micro.integrator.management.apis.Constants.NAME;
+import static org.wso2.micro.integrator.management.apis.Constants.STATUS;
 
-    private static final String CHILDREN_ATTRIBUTE = "children";
+public class EndpointResource implements MiApiResource {
 
-    public EndpointResource(String urlTemplate){
-        super(urlTemplate);
+    private static Log LOG = LogFactory.getLog(EndpointResource.class);
+    //HTTP method types supported by the resource
+    Set<String> methods;
+    // Endpoint is active property
+    private static final String IS_ACTIVE = "isActive";
+
+    public EndpointResource() {
+        methods = new HashSet<>();
+        methods.add(Constants.HTTP_GET);
+        methods.add(Constants.HTTP_POST);
     }
 
     @Override
     public Set<String> getMethods() {
-
-        Set<String> methods = new HashSet<>();
-        methods.add(Constants.HTTP_GET);
         return methods;
     }
 
     @Override
-    public boolean invoke(MessageContext messageContext) {
+    public boolean invoke(MessageContext messageContext,
+                          org.apache.axis2.context.MessageContext axis2MessageContext,
+                          SynapseConfiguration synapseConfiguration) {
 
-        buildMessage(messageContext);
-
-        org.apache.axis2.context.MessageContext axis2MessageContext =
-                ((Axis2MessageContext) messageContext).getAxis2MessageContext();
+        if (messageContext.isDoingGET()) {
+            String param = Utils.getQueryParameter(messageContext, "endpointName");
 
-        String param = Utils.getQueryParameter(messageContext, "endpointName");
-
-        if (Objects.nonNull(param)) {
-            populateEndpointData(messageContext, param);
+            if (Objects.nonNull(param)) {
+                populateEndpointData(messageContext, param);
+            } else {
+                populateEndpointList(messageContext, synapseConfiguration);
+            }
+            axis2MessageContext.removeProperty(Constants.NO_ENTITY_BODY);
```

Comment:
Shall we move this to Utils.setJsonPayLoad() method. This is missing in else part also. Instead of adding in all the places, it would be better to remove when we set a payload. wdyt

Reply:
this is only applicable for GET requests.